bundles/net.tourbook.cloud/src/net/tourbook/cloud/suunto/SuuntoRoutesUploader.java

```diff
@@ -325,27 +330,45 @@ public void run(final IProgressMonitor monitor) throws InvocationTargetException
             monitor.subTask(NLS.bind(Messages.Dialog_UploadRoutesToSuunto_SubTask,
                   UI.SYMBOL_WHITE_HEAVY_CHECK_MARK,
                   UI.SYMBOL_WHITE_HEAVY_CHECK_MARK));
+
+            monitor.worked(1);
+
+            monitor.done();
+
+            return Status.OK_STATUS;
          }
       };
 
-      try {
-         final long start = System.currentTimeMillis();
+      final long start = System.currentTimeMillis();
 
-         TourLogManager.showLogView(AutoOpenEvent.TOUR_UPLOAD);
-         TourLogManager.log_TITLE(NLS.bind(Messages.Log_UploadRoutesToSuunto_001_Start, numberOfTours));
+      TourLogManager.log_TITLE(NLS.bind(Messages.Log_UploadRoutesToSuunto_001_Start, numberOfTours));
 
-         new ProgressMonitorDialog(Display.getCurrent().getActiveShell()).run(true, true, runnable);
+      job.setPriority(Job.INTERACTIVE);
+      job.schedule();
 
-         TourLogManager.log_TITLE(String.format(LOG_CLOUDACTION_END, (System.currentTimeMillis() - start) / 1000.0));
+      job.addJobChangeListener(new JobChangeAdapter() {
+         @Override
+         public void done(final IJobChangeEvent event) {
 
-         MessageDialog.openInformation(
-               Display.getDefault().getActiveShell(),
-               Messages.Dialog_UploadRoutesToSuunto_Title,
-               NLS.bind(Messages.Dialog_UploadToursToSuunto_Message, numberOfUploadedTours[0], numberOfTours - numberOfUploadedTours[0]));
+            if (PlatformUI.isWorkbenchRunning() && event.getResult().isOK()) {
 
-      } catch (final InvocationTargetException | InterruptedException e) {
-         StatusUtil.log(e);
-         Thread.currentThread().interrupt();
-      }
+               PlatformUI.getWorkbench().getDisplay().asyncExec(() -> {
+
+                  TourLogManager.log_TITLE(String.format(LOG_CLOUDACTION_END, (System.currentTimeMillis() - start) / 1000.0));
+
+                  final String infoText = NLS.bind(Messages.Dialog_UploadToursToSuunto_Message,
+                        numberOfUploadedTours[0],
+                        numberOfTours - numberOfUploadedTours[0]);
+
+                  final NotificationPopup notication = NotificationPopup.forDisplay(Display.getCurrent())
+                        .title(Messages.Dialog_UploadRoutesToSuunto_Title, false)
+                        .text(infoText)
+                        .delay(2000)
+                        .build();
+                  notication.open();
```

Comment:
I think that the delay time of 2 seconds is a bit short, in `net.tourbook.common.UI.showStatusLineMessage(String)` a 3 second delay is used.

All `NotificationPopup`s should be put into one place, similar like in `UI.showStatusLineMessage()` with the name, e.g. `UI.showNotificationMessage()`

The delay time for statusline and popup messaages maybe could be customized in the pref page `Appearance`

Reply:
Done

![image](https://github.com/mytourbook/mytourbook/assets/25235665/3dfdcf81-13f2-459e-bb58-fb75bce6791e)